README_zh.md

```diff
@@ -0,0 +1,252 @@
+# Telegram 备份下载工具
+
+[English](./README.md)
+
+支持把消息 （格式为 [JSON](#格式)）和媒体从特定的对话、群组、频道中导出。
+
+它只能备份那些已经下载到本地的消息，如果有被中断下载的文件，将会使其恢复下载状态。
+
+此工具的工作方式就像是一个 Telegram 的客户端，所以你必须输入手机号，接着是验证码和密码（如果设置的话）
+
+此工具**不会获取频道的评论**。如果你需要评论内容的话，你必须加入频道的评论群组。
+
+## 安装
+
+```
+go get github.com/3bl3gamer/tg-history-dumper
+tg_history_dumper [args]
+```
+
+或者
+
+```
+git clone https://github.com/3bl3gamer/tg-history-dumper
+cd tg-history-dumper
+go build
+./tg_history_dumper [args]
+```
+
+## 使用方式
+
+### 准备工作
+
+`app_id` 和 `app_hash` 是必须的. 如何获取请参考 https://core.telegram.org/api/obtaining_api_id#obtaining-api-id
+
+### 最简单的使用方式
+
+```tg_history_dumper -app-id=12345 -app-hash=abcdefg```
+
+它会将获取到的凭证信息存储在 `tg.session` 文件中并且开始下载所有对话框（默认不包括媒体文件）数据到`history`文件夹。
+
+### 配置
+
+默认配置信息将会从`config.json`文件中读取。如果你起了不同的名称，可以使用`-config` [参数](#命令行参数)将其告诉此工具。
+
+格式:
+
+```json
+{
+    "app_id": 12345,
+    "app_hash": "abcdefg",
+    "socks5_proxy_addr": "127.0.0.1:9050",
+    "request_interval_ms": 1000,
+    "session_file_path": "tg.session",
+    "out_dir_path": "history",
+    "history": [
+        "all",
+        {"exclude": {"type": "channel"}},
+        {"username": "my_channel"}
+    ],
+    "media": [
+        "none",
+        {"type": "user"},
+        {"username": "my_channel"}
+    ],
+}
+```
+
+* `app_id` 和 `app_hash` — 参考 [准备工作](#准备工作);
+* `socks5_proxy_addr` — (可选) `address:post` 对应 SOCKS5 端口;
+* `request_interval_ms` — (可选, 默认值 1000) 请求历史消息数据的时间间隔（可能会减小，尽管这可能并不会加快处理速度，毕竟 TG 那边对请求频率做了限制。）
+* `session_file_path` — (可选, 默认 `tg.session`) session 文件的位置（这个文件是为了让你下次不必重新登录）
+* `out_dir_path` — (可选, 默认 `history`) 存储历史消息和媒体文件的文件夹。
+* `history` — (可选, 默认 `{"type": "user"}`) 聊天过滤[规则](#规则)。
+* `media` — (可选, 默认 `"none"`) 聊天过滤[规则](#规则)，仅在符合`history`规则的聊天中生效。
+
+如果配置中有了 `app_id` 和 `app_hash`，那么此工具就不会使用命令行参数中的那个了，你也没必要在命令行参数中再附带这两个参数。
+
+
+
+### 规则
+
+这些规则会被使用在过滤特殊的对话中（或者过滤这些对话中的媒体文件中）。
+
+如果命中（备份规则）一些规则，并且后面也没有被另一些规则反向命中（不备份的规则）掉 ，对话/文件就会被备份下来。如果没有配置任何规则将不备份任何东西。
+
+
+
+比如，这个规则就是仅备份对：
+
+```json
+"history": {"type": "user"}
+```
+
+接受所有的对话，除了频道（注意，不一定只是一个规则。多个规则是从前向后应用）：
+
+```json
+"history": [
+    "all",
+    {"exclude": {"type": "channel"}}
+]
+```
+
+接受对话和群组中的所有媒体文件，但是群组中的媒体文件有 500MiB 的大小限制：
+
+```json
+"media": [
+    {"type": "user"},
+    {"type": "group", "media_max_size": "500M"}
+]
+```
+
+接受所有对话中的媒体文件和两个群组中的媒体文件，两个群组媒体文件的大小限制是 500MiB：
+
+```json
+"media": [
+    {"type": "user"},
+    {"only": [
+        {"title": "Group A"},
+        {"title": "Group B"},
+    ], "with": {"media_max_size": "500M"}}
+]
+```
+
+`only`规则也可以这么写：
+
+```json
+{"title": "Group A", "media_max_size": "500M"},
+{"title": "Group B", "media_max_size": "500M"}
+```
+
+#### 属性的规则
+
+```json
+{
+    "id": 123,
+    "title": "Name",
+    "username": "uname",
+    "type": "user",
+    "media_max_size": "500M"
+}
+```
+
+通过下列的属性来匹配聊天和文件：
+
+* `id` 可以从 [聊天列表](#聊天列表) 获取。
+* `title` 对于用户来说是用户名。
```

Comment:
（Reverse translation is a wise approach :p ）
I hesitated for a while when I translated here that day. Because most Chinese don't use first name + last name be their name on the net. But considering the data structure of Telegram, It should indeed be translated into F + L actually.